data/api/subgraph.py

```diff
@@ -0,0 +1,61 @@
+import requests
+import diskcache
+import json
+import logging
+import os
+from statistics import mean
+from dotenv import load_dotenv
+from collections import defaultdict
+
+from model.constants import epochs_per_day, gwei, eth_deposited_per_validator
+
+load_dotenv()
+cache = diskcache.Cache(".api.cache")
+
+
+@cache.memoize(expire=(24 * 60 * 60))  # cached for 24 hours
+def get_6_month_validator_deposit_data():
+    SUBGRAPH_API_KEY = os.getenv("SUBGRAPH_API_KEY")
+    if SUBGRAPH_API_KEY:
+        API_URI = (
+            "https://gateway.thegraph.com/api/"
+            + SUBGRAPH_API_KEY
+            + "/subgraphs/id/0x540b14e4bd871cfe59e48d19254328b5ff11d820-0"
+        )
+        GRAPH_QUERY = """
+        {
+        dailyDeposits(first: 180) {
+            id
+            dailyAmountDeposited
+            }
+        }
+        """
+        try:
+            JSON = {"query": GRAPH_QUERY}
+            r = requests.post(API_URI, json=JSON)
+            return r.json().get("data", {})
+        except requests.exceptions.HTTPError as err:
+            logging.error(err)
+            return {}
+    else:
+        logging.error("SUBGRAPH_API_KEY not defined")
```

Comment:
It's likely that most people won't have the `SUBGRAPH_API_KEY` defined, so I suggest making the message a warning and saying `Variable SUBGRAPH_API_KEY not defined, using default value instead`.

Reply:
Let's add the `SUBGRAPH_API_KEY` details to the documentation in the system parameter docstring too - explaining how to set the value, or directing to relevant The Graph docs.